lib/ansible/plugins/action/synchronize.py

```diff
@@ -341,7 +341,10 @@ def run(self, tmp=None, task_vars=None):
                 # If no rsync_path is set, become was originally set, and dest is
                 # remote then add privilege escalation here.
                 if self._play_context.become_method == 'sudo':
-                    rsync_path = 'sudo rsync'
+                    if self._play_context.become_user not in ('root'):
+                        rsync_path = 'sudo -u %s rsync' % self._play_context.become_user
```

Comment:
Ok, setting now to `self._play_context.become_user` but I guess tests won't cover non-"root" users under different systems.